crates/but/src/tui/get_text.rs

```diff
@@ -60,13 +61,40 @@ fn is_terminal_dumb() -> bool {
 /// - Fall back to platform defaults (vi on Unix, notepad on Windows)
 /// - Add comprehensive unit tests for all scenarios
 fn get_editor_command() -> Result<String> {
-    get_editor_command_impl(&|key| std::env::var(key), is_terminal_dumb())
+    get_editor_command_impl(
+        &|key| std::env::var(key),
+        &|command| which(command).is_ok(),
+        is_terminal_dumb(),
+    )
+}
+
+/// Get the platform fallback editor.
+fn get_platform_editor<F>(command_exists: &F) -> String
+where
+    F: Fn(&str) -> bool,
+{
+    if cfg!(windows) {
+        "notepad".to_owned()
+    } else if cfg!(target_os = "linux") && command_exists("editor") {
+        // Linux: git source code has "vi" as fallback editor, but debian-based distros override this to
+        // "editor" which is an alternatives-managed link to the user's preferred editor.
+        // See https://sources.debian.org/src/git/1%3A2.51.0-1/debian/rules#L17
+        // We check whether "editor" is in the path and return it if it is, otherwise we stay with "vi".
+        "editor".to_owned()
+    } else {
+        "vi".to_owned()
+    }
 }
 
 /// Internal get_editor_command_implementation that can be tested without modifying environment
-fn get_editor_command_impl<F>(env_var: &F, terminal_is_dumb: bool) -> Result<String>
+fn get_editor_command_impl<F, CE>(
```

Comment:
Thinking about the function signature here - maybe it would be nice to have a something that takes an `Option`? Seeing the tests defining `doesnt_exist()` bubbled this idea